### PR TITLE
big stack on screen

### PR DESCRIPTION
**Pull request by mcmenaminadrian**

### CODE REVIEW

File: SplitFiles/control.c [hunk 1]
```diff
@@ -23,7 +73,7 @@ int buffer_push(unsigned char c, ControlBuffer* buff) {
 
 unsigned char buffer_pop(ControlBuffer* buff) {
     unsigned char c;
-    if (buff->bufferIndex == 0) {
+    if (buff->bufferIndex == 1) {
```
Comment (ConorMcPersonal) [inline SplitFiles/control.c:76]: Not at all sure I get this - what's wrong with bufferIndex 0?

File: SplitFiles/control.c [hunk 2]
```diff
@@ -11,6 +11,56 @@
...
+void buffer_restack(const ControlBuffer* const buff) {
+    int i;
+    char* command;
+    const int buffer_index = buff->bufferIndex;
+    // first clean the buffer
+    for (i = 2; i <= CONTROL_BUFFER_SIZE; i++) {
+        printf(PRINTAT"%c%c""            \0", 21, i);
+    }
+    // now write out stack
+    for (i = 2; i <= buffer_index; i++) {
```
Comment (ConorMcPersonal) [inline SplitFiles/control.c:58]: Confused - why start from i=2 but use i-1 as the index?

Comment (mcmenaminadrian) [reply SplitFiles/control.c:58]: You're supposed to comment before you merge the code!!

This index issue was at the heart of my problems. Your code inserted at 0 at the start but only inserted at 1 thereafter. Trying to make it consistent so it started from 1 at all times.

Comment (mcmenaminadrian) [reply SplitFiles/control.c:58]: The i=2 was simply a convenience for the PRINTAT though

Comment (ConorMcPersonal) [reply SplitFiles/control.c:58]: Was hardly going to get snotty over a PR at this stage - not like I "own" the project.

This is the insert (push) code:
    (buff->buffer)[buff->bufferIndex] = c;
    buff->bufferIndex = buff->bufferIndex + 1;
    (buff->buffer)[buff->bufferIndex] = 0;
bufferIndex started out at 0, so the first character went there.  Reason for the third line above was to ensure we always had a string terminator so printing the buffer would stop at the right place (the format added spaces thereafter).

**Pull request merged**